src/Integration.Vsix.UnitTests/Commands/ProjectExcludePropertyToggleCommandTests.cs

```diff
@@ -0,0 +1,355 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="ProjectExcludePropertyToggleCommandTests.cs" company="SonarSource SA and Microsoft Corporation">
+//   Copyright (c) SonarSource SA and Microsoft Corporation.  All rights reserved.
+//   Licensed under the MIT License. See License.txt in the project root for license information.
+// </copyright>
+//-----------------------------------------------------------------------
+
+using Microsoft.VisualStudio.ComponentModelHost;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using SonarLint.VisualStudio.Integration.Vsix;
+using System;
+using System.Windows.Threading;
+
+namespace SonarLint.VisualStudio.Integration.UnitTests.Commands
+{
+    [TestClass]
+    public class ProjectExcludePropertyToggleCommandTests
+    {
+        #region Test boilerplate
+
+        private ConfigurableVsProjectSystemHelper projectSystem;
+        private IServiceProvider serviceProvider;
+
+        [TestInitialize]
+        public void TestInitialize()
+        {
+            var provider = new ConfigurableServiceProvider();
+            this.projectSystem = new ConfigurableVsProjectSystemHelper(this.serviceProvider);
+            provider.RegisterService(typeof(IProjectSystemHelper), this.projectSystem);
+
+            var host = new ConfigurableHost(provider, Dispatcher.CurrentDispatcher);
+            var propertyManager = new ProjectPropertyManager(host);
+            var mefExports = MefTestHelpers.CreateExport<IProjectPropertyManager>(propertyManager);
+            var mefModel = ConfigurableComponentModel.CreateWithExports(mefExports);
+            provider.RegisterService(typeof(SComponentModel), mefModel);
+
+            this.serviceProvider = provider;
+        }
+
+        #endregion
+
+        #region Tests
+
+        [TestMethod]
+        public void ProjectExcludePropertyToggleCommand_Ctor()
+        {
+            Exceptions.Expect<ArgumentNullException>(() => new ProjectExcludePropertyToggleCommand(null));
+        }
+
+        [TestMethod]
+        public void ProjectExcludePropertyToggleCommand_Invoke_SingleProject_TogglesValue()
+        {
+            // Setup
+            OleMenuCommand command = CommandHelper.CreateRandomOleMenuCommand();
+            command.Enabled = true;
+
+            var testSubject = new ProjectExcludePropertyToggleCommand(serviceProvider);
```

Comment:
Might worth documenting that in practice the value is set to true or will be cleared when using this command, in case it's not documented already.